src/juju/watchers.ts

```diff
@@ -86,7 +87,13 @@ export function processDeltas(
     if (delta[0] === DeltaEntityTypes.ACTION) {
       _process(ReduxDeltaEntityTypes.ACTIONS, delta[2].id);
     } else if (delta[0] === DeltaEntityTypes.APPLICATION) {
-      _process(ReduxDeltaEntityTypes.APPLICATIONS, delta[2].name);
+      const formatted: ApplicationData = {
+        [delta[2].name]: delta[2],
```

Comment:
These would probably be a lot easier to follow with some more descriptive names. You could use a destructure in the forEach, something like `deltas.forEach(([action, change, delta]) => {`.

Reply:
Unfortunately I'm not able to destructure due to a bug in TS. The discussion is in one of the previous PR's here.